Added support for custom length in cloudwatch components (#4385)
grafana/agent: docs/sources/static/configuration/integrations/cloudwatch-exporter-config.md

  # Required: List of statistic types, e.g. "Minimum", "Maximum", etc.
  statistics: [ <string> ]

  # Optional: See the `period and length` section below.
  period: [ <duration> | default = 5m ]

  # Optional: See the `period and length` section below.
  length: [ <duration> | default = calculated based on `period` ]
```

### Period and length

> `period` controls primarily the width of the time bucket used for aggregating metrics collected from
> CloudWatch. `length`
> controls how far back in time CloudWatch metrics are considered during each agent scrape. If both settings are
> configured,
> the time parameters when calling CloudWatch APIs works as follows:

Reviewer: ```suggestion
`period` controls the width of the time bucket used for aggregating metrics collected from CloudWatch.
`length` controls how far back in time CloudWatch metrics are considered during each agent scrape.
If both settings are configured, the time parameters when calling CloudWatch APIs work as follows:
```

Tidying up the line breaks

![](https://grafana.com/media/docs/agent/cloudwatch-period-and-length-time-model-2.png)

> As noted above, if across multiple metrics under the same static or discovery job, there's different `period`
> and/or `length`
> the minimum of all periods, and maximum of all lengths is configured.

Reviewer: ```suggestion
As noted above, if there is a different `period` or `length` across multiple metrics under the same static or discovery job, 
the minimum of all periods, and maximum of all lengths is configured.
```

> On the other hand, if `length` is not configured, both period and length settings will be calculated based on the
> required
> `period` configuration attribute.

Reviewer: ```suggestion
On the other hand, if `length` is not configured, both period and length settings are calculated based on 
the required `period` configuration attribute.
```

If all metrics within a job (discovery or static) have the same `period` value configured, CloudWatch APIs will be

> requested
> for metrics from the scrape time, to `period`s seconds in the past. The values of these are exported to Prometheus.

Reviewer: ```suggestion
requested for metrics from the scrape time, to `period`s seconds in the past. 
The values of these metrics are exported to Prometheus.
```
